ebos/eclgenericwriter_impl.hh

```diff
@@ -381,38 +461,46 @@ exportNncStructure_(const std::unordered_map<int,int>& cartesianToActive, const
             if ( cc2 < cc1 )
                 std::swap(cc1, cc2);
 
-            auto cellDiff = cc2 - cc1;
-
             // Re-ordering in case of non-empty mapping between equilGrid to grid
             if (map) {
                 c1 = map(c1); // equilGridToGrid map
                 c2 = map(c2);
             }
 
-            if (cellDiff != 1 &&
-                cellDiff != nx &&
-                cellDiff != nx*ny &&
-                !directVerticalNeighbors(cartDims, cartesianToActive, cc1, cc2)) {
-                // We need to check whether an NNC for this face was also specified
-                // via the NNC keyword in the deck (i.e. in the first origNncSize entries.
-                auto t = globalTrans().transmissibility(c1, c2);
-                auto candidate = std::lower_bound(nncData.begin(), nncData.end(), NNCdata(cc1, cc2, 0.0));
-
-                while ( candidate != nncData.end() && candidate->cell1 == cc1
-                     && candidate->cell2 == cc2) {
+            if (isNumAquConn(cc1, cc2) ||
+                ! (isCartesianNeighbour(cc1, cc2) ||
+                   directVerticalNeighbors(cartDims, cartesianToActive, cc1, cc2)))
```

Comment:
I misread this at first, missing the parenthesis. Suggest pulling the second condition out as a separate variable for readability:
```
const bool cartesianOrDirevtVertNb = isCartesianNeighbour(cc1, cc2) || directVerticalNeighbors(cartDims, cartesianToActive, cc1, cc2);
if (!cartesianOrDirectVertNb || isNumAquConn(cc1, cc2)) {
...
}
```
(Here I flipped the order in the if to use the already calculated value first in the logical shortcutting).
```

Reply:
> Here I flipped the order in the if to use the already calculated value first in the logical shortcutting

Sure, but to be honest I was actually using short-circuit behaviour to avoid running `directVerticalNeighbors()` unless really needed.  I'll think of an alternative.

Reply:
> to be honest I was actually using short-circuit behaviour to avoid running `directVerticalNeighbors()` unless really needed. I'll think of an alternative.

Good point, as long as the result is easy to understand I am fine with it.

Reply:
> as long as the result is easy to understand I am fine with it.

I've extracted that part of the condition into a new helper (lambda) named `isDirectNeighbours()`.